pkg/provision/storage/asyncstorage/errors.go

```diff
@@ -0,0 +1,17 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package asyncstorage
+
+import "errors"
+
+var NotReadyError = errors.New("async storage component is not ready")
```

Comment:
The error value here is just to distinguish between an unexpected error and a routine error (i.e. where we just need to requeue/wait). This is different from the `NotReadyError` in the storage package, which has additional context (like message and requeue duration). The storage package `NotReadyError` is part of the interface as it's read by the caller to determine information about how to requeue and set messages. 

The way to remove the three checks would be break out storage errors into a separate package, which is IMO messier than just doing standard golang copy-paste error handling. We could set the message and requeue delay in the asyncstorage package, but it would mean a circular import by default.

Reply:
I'm OK with keeping as is for time being.